packages/ethersjs-provider/src/EthersProvider.ts

```diff
@@ -58,17 +58,43 @@ export class EthersProvider extends ethers.providers.BaseProvider
     concurrency: number
   ) {
     super(provider.getNetwork());
+    this.on('block', (blockNumber) => this._mostRecentLatestBlockHeaders = null);
     this.provider = provider;
     this.performQueue = queue(
       (item: PerformQueueTask, callback: (err: Error, results: any) => void) => {
         const _this = this;
         retry(
           { times, interval },
           async () => {
-            if (item.message === "send") {
-              return await _this.providerSend(item.params.method, item.params.params);
+            if (item.message === 'send') {
+              const {method, params: [blocktag, includeHeaders]} = item.params;
```

Comment:
can this all go in a function so the core queue processing is still easy to read?